plugins/module_utils/identity/keycloak/keycloak.py

```diff
@@ -1694,3 +1749,95 @@ def delete_component(self, cid, realm='master'):
         except Exception as e:
             self.module.fail_json(msg='Unable to delete component %s in realm %s: %s'
                                       % (cid, realm, str(e)))
+
+    def create_user(self, userrep, realm='master'):
+        """ Create a Keycloak user.
+
+        :param userrep: a UserRepresentation of the user to be created. Must contain at minimum the field name.
+        :return: HTTPResponse object on success
+        """
+        users_url = URL_USERS.format(url=self.baseurl, realm=realm)
+        try:
+            return open_url(users_url, method='POST', headers=self.restheaders,
+                            data=json.dumps(userrep), validate_certs=self.validate_certs)
+        except Exception as e:
+            self.module.fail_json(msg="Could not create user %s in realm %s: %s"
+                                      % (userrep['user_username'], realm, str(e)))
+
+    def get_users_by_username(self, username, realm='master'):
+        """ Fetch the name and ID of user by username on the Keycloak server.
+
+        To fetch the full data of the user.
+
+        :param username: The username of the user. A lookup will be performed to retrieve the user ID.
+        :param realm: Return the groups of this realm (default "master").
+        """
```

Comment:
Why does this need to return a list (and the received thus have a lot of logic around the response, and so forth)?

Surely, keycloak only allows one user with a username per realm, so the response could be None or the user object rather than a list of objects.

The calling code could then be reduced to:

```
    found_user = kc.get_user_by_username(username, realm=realm)
    before_user = {} if not found_user else found_user
```